docs/getting-started/cli.md

```diff
@@ -0,0 +1,198 @@
+
+To share your rules by adding them to the Semgrep Registry:
+
+```console
+semgrep publish <path/to/rules>
+```
+
+### Scan without sending results to Semgrep
+
+To scan your project **without** sending data to Semgrep, use:
+
+```console
+semgrep ci --dry-run
+```
+
+This can be helpful to verify the results of a specific ruleset or to see how your findings change based on the rulesets you choose for your scans.
+
+### Scan using Semgrep's OSS engine
+
+To scan your project with Semgrep's OSS Engine, even though you have the Pro Engine enabled in Semgrep Cloud Platform:
+
+```console
+semgrep ci --oss-only
+```
+
+:::info
+See [Semgrep Pro versus Semgrep OSS](/semgrep-pro-vs-oss) for information on the differences between Semgrep Pro and OSS Engines.
+:::
+
+## Scan using specific Semgrep Products
+
+When you run `semgrep ci`, you scan your project with Semgrep Code, Semgrep Supply Chain, and Semgrep Secrets. To scan your project with just one product, run:
+
+```console
+# scan with Semgrep Code
+semgrep ci --code
+
+# scan with Semgrep Supply Chain
+semgrep ci --supply-chain
+
+# scan with Semgrep Secrets
+semgrep ci --secrets
+```
+
+## Extend timeout thresholds
+
+Depending on the file sizes in your project, you may need to increase the timeout threshold so that Semgrep doesn't time out before the scan completes. You can control this value using the `--timeout-threshold` flag, which refers to the maximum amount of time Semgrep spends scanning a single file. The default value is 30 seconds. Semgrep attempts to scan each file with this timeout value three times.
+
+```console
+# increase timeout to 45 seconds
+semgrep ci --timeout-threshold 45
+```
+
+## Improve performance for large codebases
+
+You can set the number of subprocesses Semgrep uses to run checks in parallel:
+
+```console
+semgrep scan -j NUMBER_OF_SUBPROCESSES
+```
+
+By default, Semgrep defaults to the number of cores detected on the system on which Semgrep is running, but `-j = 1` if you're passing in `--pro`. For additional information, see [Parallelization](/kb/semgrep-code/scan-engine-kill)
+
+## Set log levels
+
+Semgrep provides three levels of logging:
+
+| **Log level** | **Flag** | **Description** |
+| - | - | - |
+| Default | None | Prints scan progress, findings information, warnings, and errors. |
+| Verbose | `-v` or `--verbose` | Includes everything printed when using the default logging level, adding a list of rules and details such as skipped files. |
+| Debug | `--debug` | Logs the entire scan process at a high level of detail. |
+
+### Example usage
+
+To set the logging level for a scan, include the flag when scanning your project:
+
+```console
+# run a scan and get debug logs
+semgrep ci --debug
+```
+
+## Exit codes
+
+The CLI commands `semgrep ci` and `semgrep scan` finish with exit code `0` as long as the scan completes, regardless of whether there were findings. To finish with exit code `1` when there are findings, pass in the `--error` flag.
```

Comment:
This is not true. If you've configured blocking rules in SCP, `semgrep ci` will exit 1.